core/meta/src/TCling.cxx

```diff
@@ -5283,8 +5283,8 @@ void TCling::UpdateClassInfoWithDecl(const void* vTD)
       }
       clang::QualType type( td->getTypeForDecl(), 0 );
 
-      auto declName=ND->getNameAsString().c_str();
-      if (!TClass::HasNoInfoOrEmuOrFwdDeclaredDecl(declName)){
+      auto declName = ND->getNameAsString();
```

Comment:
All I'm saying is that this "fix" will introduce memory churn because you end up constructing a new string. You might want to make sure that does not create other issues.